src/main/java/com/softserve/cli/impl/OptimizedPaintingFence.java

```diff
@@ -0,0 +1,45 @@
+package com.softserve.cli.impl;
+
+import java.util.Scanner;
+
+import com.softserve.algorithms.Algorithms;
+import com.softserve.cli.AbstractCLI;
+import com.softserve.cli.CommandLineInterface;
+import com.softserve.cli.InvalidInputHandler;
+import com.softserve.cli.util.InputUtil;
+
+public class OptimizedPaintingFence extends AbstractCLI implements CommandLineInterface{
+	
+	private int numberOfColors;
+    private int numberOfPosts;
+
+	public OptimizedPaintingFence(Scanner prompt) {
+		super(prompt);
+	}
+
+	@Override
+	public void run() {
+		System.out.printf("%s\n%s\n", algorithmOption.getOptionName(),
+                algorithmOption.getOptionDescription());
+        InvalidInputHandler.promptIfInvalidValue(this::promptNParamethers, INVALID_NUMBER_MSG);
+        InvalidInputHandler.promptIfInvalidValue(this::promptKParamethers, INVALID_NUMBER_MSG);
+        System.out.println("The MAX number of possible ways of painting the fence is " + Algorithms.findingMaxWaysOfPaintingTheFence(numberOfPosts, numberOfColors));
+	}
+	
+	private boolean promptNParamethers() {
```

Comment:
Rename this method to promptNumberOfColors

Reply:
> Rename this method to promptNumberOfColors

solved